toolchain/check/context.h

```diff
@@ -208,22 +212,32 @@ class Context {
   auto AddNameToLookup(SemIR::NameId name_id, SemIR::InstId target_id) -> void;
 
   // Performs name lookup in a specified scope for a name appearing in a
-  // declaration, returning the referenced instruction. If scope_id is invalid,
-  // uses the current contextual scope.
+  // declaration. If found, returns the referenced instruction and false. If
+  // poisoned, returns invalid instructions and true. poisoned. If scope_id is
+  // invalid, uses the current contextual scope.
+  // TODO: For poisoned names, return the poisoning instruction.
   auto LookupNameInDecl(SemIR::LocId loc_id, SemIR::NameId name_id,
-                        SemIR::NameScopeId scope_id) -> SemIR::InstId;
+                        SemIR::NameScopeId scope_id)
+      -> std::pair<SemIR::InstId, bool>;
 
   // Performs an unqualified name lookup, returning the referenced instruction.
   auto LookupUnqualifiedName(Parse::NodeId node_id, SemIR::NameId name_id,
                              bool required = true) -> LookupResult;
 
+  struct LookupNameInExactScopeResult {
+    SemIR::InstId inst_id;
+    SemIR::AccessKind access_kind;
+    bool is_poisoned = false;
```

Comment:
Maybe add comments to the members? Note this is sort of an alternative to the documentation on LookupNameInExactScope (which I note doesn't explain `access_kind`, but that's a pre-existing issue).

```suggestion
  struct LookupNameInExactScopeResult {
    // The matching entity if found, or invalid if poisoned or not found.
    SemIR::InstId inst_id;
    // The access level required to use inst_id, if it's valid.
    SemIR::AccessKind access_kind;
    // Whether a poisoned entry was found.
    bool is_poisoned = false;
```

Or if you'd rather keep documentation on the function, maybe just add a note of where to look:

```suggestion
  // See LookupNameInExactScope for return information.
  struct LookupNameInExactScopeResult {
    SemIR::InstId inst_id;
    SemIR::AccessKind access_kind;
    bool is_poisoned = false;
```

Reply:
Added comments to the members, thanks!